src/bin/pg_autoctl/cli_show.c

```diff
@@ -963,41 +1019,82 @@ cli_show_uri(int argc, char **argv)
 		{
 			bool monitorDisabledIsOk = false;
 
-			if (!keeper_config_read_file_skip_pgsetup(&kconfig,
+			if (!keeper_config_read_file_skip_pgsetup(kconfig,
 													  monitorDisabledIsOk))
 			{
 				/* errors have already been logged */
 				exit(EXIT_CODE_BAD_CONFIG);
 			}
 
-			if (!monitor_init(&monitor, kconfig.monitor_pguri))
+			if (!monitor_init(monitor, kconfig->monitor_pguri))
 			{
 				/* errors have already been logged */
 				exit(EXIT_CODE_BAD_CONFIG);
 			}
-			ssl = &(kconfig.pgSetup.ssl);
+
+			*ssl = kconfig->pgSetup.ssl;
 			break;
 		}
 
 		default:
 		{
 			log_fatal("Unrecognized configuration file \"%s\"",
-					  kconfig.pathnames.config);
+					  kconfig->pathnames.config);
 			exit(EXIT_CODE_INTERNAL_ERROR);
 		}
 	}
+}
+
+
+/*
+ * cli_show_uri prints the URI to connect to with psql.
+ */
+static void
+cli_show_uri(int argc, char **argv)
+{
+	KeeperConfig kconfig = keeperOptions;
+	Monitor monitor = { 0 };
+	SSLOptions ssl = { 0 };
+
+	/*
+	 * We are given either --monitor postgres://uri or --pgdata; in the first
+	 * case we just connect to that URI, in the second case we read the monitor
+	 * URI's from the local configuration file.
+	 */
+	if (!IS_EMPTY_STRING_BUFFER(kconfig.monitor_pguri))
+	{
+		if (!monitor_init(&monitor, kconfig.monitor_pguri))
+		{
+			/* errors have already been logged */
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+		/* cook the sslmode=prefer bits */
+		ssl.active = true;
+		ssl.sslMode = SSL_MODE_PREFER;
```

Comment:
Any chance we could get this from the `monitor_pguri`? i.e. if monitor uri has `sslmode=verify-full&sslrootcert=somecert.crt` we would use those options.

Reply:
Done now.